### PR TITLE
Dockerhub publish CI

### PR DESCRIPTION
**Pull request by Mropat**
This PR adds/fixes ...

CI workflows to build container 

## Review
- [x] code approved by
- [x] tests executed by
- [x] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a
- [x] **PATCH**

## Implementation Plan
- [ ] Document in ...
- [ ] Deploy this branch
- [ ] Inform to ...

### CODE REVIEW

**Comment by hassanfa** [comment]
To get branch name try this:

```
jobs:
  main:
    name: Docker image build push 
    runs-on: ubuntu-18.04
    strategy:
      fail-fast: true
      matrix: 
        container-name: [align_qc, annotate, coverage_qc, varcall_cnvkit, varcall_py27, varcall_py36]
    steps:
      - name: Set up QEMU
        uses: docker/setup-qemu-action@v1
      - name: Set up Docker Buildx
        uses: docker/setup-buildx-action@v1
      - name: Get branch name
        id: get_branch_name
        shell: bash
        run: echo "##[set-output name=branch;]$(echo ${GITHUB_REF##*/})"
      - name: Login to DockerHub
        id: docker_login
        uses: docker/login-action@v1
        with:
          username: ${{ secrets.DOCKER_USERNAME }}
          password: ${{ secrets.DOCKER_PASSWORD }}
      - name: Build and push
        id: docker_build_push
        uses: docker/build-push-action@v2
        with:
          push: true
          tags: clinicalgenomics/balsamic:${{ steps.get_branch_name.outputs.branch }}-${{ matrix.container-name }} 
          build-args:
            CONTAINER_NAME=${{ matrix.container-name }}
```

**Comment by hassanfa** [comment]
Also, I recommend not using some thirdpart container github action. Docker is a verified creator and they have their own action: https://github.com/marketplace/actions/build-and-push-docker-images

This works well for Vogue and BALSAMIC.

File: .github/workflows/build_container_pull.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+---
+
+name: Build container for stage testing
+on: [push]
```
Comment (moonso) [inline .github/workflows/build_container_pull.yml:4]: Do we really want to build a new container on every push to every branch?

File: .github/workflows/build_container_pull.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+      - name: Output branch name
+        run: echo ${BRANCH_NAME}
+
+      - name: Build and publish to Docker Hub
+        uses: elgohr/Publish-Docker-Github-Action@master
+        with:
+          name: clinicalgenomics/cg-service-stage
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+          tags: "${{ env.BRANCH_NAME }}"
```
Comment (moonso) [inline .github/workflows/build_container_pull.yml:27]:
```suggestion
          tags: "${{ env.BRANCH_NAME }}"
          
```

**Review by moonso — COMMENTED** [review]
Looks good but I had one question

File: .github/workflows/build_container_pull.yml (hunk 1 above)
Comment (Mropat) [reply .github/workflows/build_container_pull.yml]: changed to pull request

**Pull request merged**